src/sentry/search/events/filter.py

```diff
@@ -1367,12 +1371,16 @@ def _default_filter_converter(self, search_filter: SearchFilter) -> Optional[Whe
         }:
             value = int(to_timestamp(value)) * 1000
 
+        if name in {"trace.span", "trace.parent_span"}:
```

Comment:
I noticed some code duplication down in the `QueryFilter` class and I decided to add the check here as well. Submitting a query with `trace.span` didn't seem to hit this code so I need to loop back and learn what this class is for and when it's called

Reply:
This is when the code runs with SnQL enabled, you can turn this on locally by enabling the `organizations:discover-use-snql` feature flag